### ISSUE TITLE
sails-mysql: Error: Consistency violation: expected `create` adapter method to send back the created record

### ISSUE THREAD

**Issue by garrettjoecox** [balderdashy/sails#4002]
**Sails version**: 1.0.0-27
**Node version**: 7.4.0
**NPM version**: 4.0.5
**Operating system**: MacOS Sierra (10.12.3)

<hr/>

This issue has been driving me nuts trying to upgrade my application to v1.. It happens very intermittently when creating a new instances that have associations on the given model. However it always happens on your first creation if you set the `migrate: "drop"` setting, and lift sails. (So each time you lift sails, your first creation should throw)

Stack Trace:
```
Error: Consistency violation: expected `create` adapter method to send back the created record b/c `fetch: true` was enabled.  But instead, got: undefined
       at _afterTalkingToAdapter (/sails-issue-consistency-violation/node_modules/waterline/lib/waterline/methods/create.js:300:25)
       at Object.success (/sails-issue-consistency-violation/node_modules/sails-mysql/lib/adapter.js:140:18)
       at afterMaybeArtificiallyWaiting (/sails-issue-consistency-violation/node_modules/machine/lib/private/intercept-exit-callbacks.js:385:21)
       at maybeArtificiallyWait (/sails-issue-consistency-violation/node_modules/machine/lib/private/intercept-exit-callbacks.js:201:20)
       at afterPotentiallyCaching (/sails-issue-consistency-violation/node_modules/machine/lib/private/intercept-exit-callbacks.js:221:11)
       at _cacheIfAppropriate (/sails-issue-consistency-violation/node_modules/machine/lib/private/intercept-exit-callbacks.js:79:18)
       at Function._interceptExit [as success] (/sails-issue-consistency-violation/node_modules/machine/lib/private/intercept-exit-callbacks.js:92:9)
       at releaseCb (/sails-issue-consistency-violation/node_modules/sails-mysql/helpers/create.js:204:26)
       at Object.success (/sails-issue-consistency-violation/node_modules/sails-mysql/helpers/private/connection/release-connection.js:37:14)
       at afterMaybeArtificiallyWaiting (/sails-issue-consistency-violation/node_modules/machine/lib/private/intercept-exit-callbacks.js:385:21)
       at Immediate.<anonymous> (/sails-issue-consistency-violation/node_modules/machine/lib/private/intercept-exit-callbacks.js:212:22)
       at runCallback (timers.js:651:20)
       at tryOnImmediate (timers.js:624:5)
       at processImmediate [as _immediateCallback] (timers.js:596:5)
```

Reproduction Repo:
https://github.com/garrettjoecox/sails-issue-consistency-violation
(Diff from generated repo are in last commit)

Reproduction Steps:
* configure sails-mysql datastore in local.js
* lift sails
* Create a few users with the route `POST /user`, and an email.
* Observe the error in the console, and that the user was in fact created but not fetched.

UPDATE:
The `meta({ fetch: true })` functionality doesn't seem to be working properly. This is likely the cause for the issue above. I pushed up another commit to the repo to demonstrate. Steps to reproduce weirdness:
* After following the steps from above, copy the ID from one of the created users
* Create 10 or 20 repos using the route `POST /repo` with the `owner` property set to the copied User ID
* Observe the logs, how the provided repo ID's are different from the fetched repo ID's

**Comment by particlebanana** [comment]
@garrettjoecox looking into it.

**Comment by particlebanana** [comment]
@garrettjoecox looks like in the mysql adapter we are relying on the PK being returned as `lastInsertId` which obviously won't work in this case because it's a custom string primary key. I'll get a patch for that out today for you.

**Comment by particlebanana** [comment]
@garrettjoecox should be fixed in sails-mysql `1.0.0-8`. If you could give it a spin and let me know if that clears it up. Install by doing `npm install sails-mysql@beta`.

**Comment by garrettjoecox** [comment]
@particlebanana Everything seems to be working as expected. Thanks a bunch!
